### PR TITLE
[USETUP] Provide Status values when on fail

### PR DESCRIPTION
**Pull request by Fraizeraust**
## Purpose
When a certain operation fails, print a NTSTATUS value for it in the debug output.
## TODO
- [ ] Fix a compile error

Cc @HBelusca

### CODE REVIEW

File: base/setup/usetup/usetup.c [hunk 1]
```diff
@@ -595,15 +595,18 @@ UpdateKBLayout(VOID)
...
     LPCWSTR pszNewLayout;
+    NTSTATUS Status;
 
     pszNewLayout = MUIDefaultKeyboardLayout();
 
     if (LayoutList == NULL)
     {
-        LayoutList = CreateKeyboardLayoutList(SetupInf, DefaultKBLayout);
-        if (LayoutList == NULL)
+        Status = CreateKeyboardLayoutList(SetupInf, DefaultKBLayout);
```
Comment (binarymaster) [inline base/setup/usetup/usetup.c]:
```
../../base/setup/usetup/usetup.c: In function 'UpdateKBLayout':
../../base/setup/usetup/usetup.c:604:16: error: assignment makes integer from pointer without a cast [-Werror]
```

Not all return values are NTSTATUSes.

https://git.reactos.org/?p=reactos.git;a=blob;f=base/setup/usetup/settings.h;hb=12654d28522a70618de162912f6a97c3e59c0ff1#l52

Comment (Fraizeraust) [reply base/setup/usetup/usetup.c]: `InfFile` is declared as *HINF* which is a handle to an INF file. Hm. Shall I use `HRESULT` as replacement for `NTSTATUS` instead?

Comment (ThFabba) [reply base/setup/usetup/usetup.c]: No. NTSTATUS values won't magically appear. You have to make those functions actually return NTSTATUS values (and change their return type).

**Comment by ThFabba** [comment]
Returning NTSTATUS is good practice, but two considerations:
* Some usetup functions mirror similar Win32 functions, and therefore should keep the same prototypes (I don't have a list, you'd have to search for similarly named APIs and check on a case-by-case basis)
* Refactoring large amounts of code is a source of regressions, so if there is no functional benefit to it, it's best to leave things alone.

If you want to make it easier to debug specific issues, perhaps add DPRINTs in the failure cases of the individual functions and print the error at the source?

**Comment by Fraizeraust** [comment]
@HBelusca Oops! Sorry about that then. I'll close close the PR then and I'll let you decide what you're going to do with it after.

**Pull request closed without merging**